modules/security-api/src/main/java/org/apache/ignite/internal/security/authentication/AuthenticationException.java

```diff
@@ -15,15 +15,15 @@
  * limitations under the License.
  */
 
-package org.apache.ignite.security;
+package org.apache.ignite.internal.security.authentication;
 
+import org.apache.ignite.internal.lang.IgniteInternalCheckedException;
 import org.apache.ignite.lang.ErrorGroups.Authentication;
-import org.apache.ignite.lang.IgniteException;
 
 /**
  * The general authentication exception.
  */
-public class AuthenticationException extends IgniteException {
+public class AuthenticationException extends IgniteInternalCheckedException {
```

Comment:
1. This exception should be public, it can be thrown from public APIs.
2. Let's use separate error codes, I see at least two: `UNKNOWN_AUTHENTICATION_TYPE` (use in `ClientInboundMessageHandler.createAuthenticationRequest`), `INVALID_CREDENTIALS`.